index.js

```diff
@@ -13,17 +15,21 @@ module.exports = function (pid, signal, callback) {
         exec('taskkill /pid ' + pid + ' /T /F', callback);
         break;
     case 'darwin':
-        buildProcessTreeDarwin(pid, tree, pidsToProcess, function () {
-            killAll(tree, signal, callback);
-        });
-        break;
-    case 'sunos':
-        buildProcessTreeSunOS(pid, tree, pidsToProcess, function () {
+        buildProcessTree(pid, tree, pidsToProcess, function (parentPid) {
+          return spawn('pgrep', ['-P', parentPid]);
+        }, function () {
             killAll(tree, signal, callback);
         });
         break;
+    // case 'sunos':
+    //     buildProcessTreeSunOS(pid, tree, pidsToProcess, function () {
+    //         killAll(tree, signal, callback);
+    //     });
+    //     break;
     default: // Linux
-        buildProcessTree(pid, tree, pidsToProcess, function () {
+        buildProcessTree(pid, tree, pidsToProcess, function (parentPid) {
+          return spawn('ps', ['-o', 'pid', '--no-headers', '--ppid', parentPid]);
+        }, function () {
```

Comment:
Since new line 31 is the exact same command as old line 70, the effect should be exactly the same except for the extra function anonymous function in the call stack.